lib/improver/spotdata/read_input.py

```diff
@@ -174,7 +196,8 @@ def data_from_dictionary(dictionary_data, key):
 
```

Comment:
I have removed data_from_dictionary and updated the unit tests to expect the normal keyerror that is raised when querying for invalid keys.